usr/src/lib/libfakekernel/common/ksid.c

```diff
@@ -0,0 +1,68 @@
+/*
```

Comment:
Doesn't this clash with what libfksmbsrv is doing with creds, sids, and ksids?   I think we may need to keep this too in the consumers, rather than in libfakekernel (leaving this ignorant of creds or ksids).

Reply:
I don't think so.  I don't see these functions defined there, anyway.  And besides, there's already some cred stuff in libfakekernel.

Reply:
I know, but I'm starting to think that was a mistake.

In general, I'd rather "promote" things into libfakekernel only after we're
fairly sure they're generic enough to satisfy the needs of all consumers.
That's not at all obvious to me in the case of cred the stuff...